contracts/staking/MultiFeeDistribution.sol

```diff
@@ -446,16 +466,14 @@ contract MultiFeeDistribution is IMultiFeeDistribution, ReentrancyGuard, Ownable
         emit Recovered(tokenAddress, tokenAmount);
     }
 
-    /* ========== MODIFIERS ========== */
-
-    modifier updateReward(address account) {
+    function _updateReward(address account) internal {
         address token = address(stakingToken);
         uint256 balance;
         Reward storage r = rewardData[token];
         uint256 rpt = _rewardPerToken(token, lockedSupply);
         r.rewardPerTokenStored = rpt;
         r.lastUpdateTime = lastTimeRewardApplicable(token);
-        if (account != address(0)) {
+        if (account != address(this)) {
```

Comment:
Why `address(this)` ?

Reply:
By adjusting the sequence of operations in the mint function, it is now not possible for this contract to have a nonzero staking balance.  So this if branch saves gas when minting to the staking contract.

I'm not sure why the original address(0) because as far as i can see that could never evaluate falsely. Maybe a leftover from the original codebase?